docs/guide/deployment.md

```diff
@@ -145,7 +145,7 @@ Typically, your static website will be hosted on https://yourUserName.gitlab.io/
 
 module.exports = {
   publicPath: process.env.NODE_ENV === 'production'
-    ? '/yourProjectName/'
+    ? '/' + process.env.CI_PROJECT_NAME + '/'
```

Comment:
Thank you for working on this!

While I agree with the changing hardcoded value to predefined GitLab CI variable, this also requires removing  a comment above the export:
```
// make sure you update `yourProjectName` with the name of your GitLab project
```

Also, it would be nice to explain than the project name is a predefined variable in GitLab CI.

Reply:
ok, you're right. thanks for taking the time to review this, I've added some explanations and a link to gitlab docs

Reply:
By the way, I would advice on using gzip compression on gitlab pages.
It's just one line to add in the `.gitlab-ci.yml` file, see https://docs.gitlab.com/ee/user/project/pages/introduction.html#serving-compressed-assets
Do you think I could do this in this PR or open a new one?

Reply:
Oh, I would agree on using gzip! Let's add it to this PR